pytorch_lightning/trainer/trainer.py

```diff
@@ -709,6 +792,27 @@ def predict(
         return_predictions: Optional[bool] = None,
         ckpt_path: Optional[str] = None,
     ) -> Optional[_PREDICT_OUTPUT]:
+        try:
+            return self._predict_impl(model, dataloaders, datamodule, return_predictions, ckpt_path)
+        except KeyboardInterrupt:
+            rank_zero_warn("Detected KeyboardInterrupt, attempting graceful shutdown...")
+            # user could press Ctrl+c many times... only shutdown once
+            if not self.interrupted:
+                self.state.status = TrainerStatus.INTERRUPTED
+                self.on_keyboard_interrupt()
+                # same treatment as below
+                self.accelerator.teardown()
+        except BaseException:
+            self.state.status = TrainerStatus.INTERRUPTED
+            if distributed_available() and self.world_size > 1:
+                # try syncing remaing processes, kill otherwise
+                self.training_type_plugin.reconciliate_processes(traceback.format_exc())
+            # give accelerators a chance to finish
+            self.accelerator.teardown()
+            self._on_exception()
+            # reset bookkeeping
+            self.state.stage = None
+            raise
         r"""
         Run inference on your data.
         This will call the model forward function to compute predictions. Useful to perform distributed
```

Comment:
Looks like the `predict_impl` function is missing

Reply:
good catch